### ISSUE TITLE
Typecheck fails for Architect create-remix-app template

### ISSUE THREAD

**Issue by lpsinger** [remix-run/remix#6296]
### What version of Remix are you using?

1.16.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

1. Create an app using `create-remix`.
2. Select `Just the basics`, `(AWS Lambda)`, `TypeScript`, and let it run `npm install`.
3. Descend into the newly created `my-remix-app` directory directory and run `npm run typecheck`.

### Expected Behavior

TypeScript type checks should complete without any errors.

### Actual Behavior

```
$ npx create-remix@latest
? Where would you like to create your app? ./my-remix-app
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix App Server if you're unsure; it's 
easy to change deployment targets. Architect (AWS Lambda)
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
⠦ Creating your app…(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠏ idealTree:my-remix-app: sill idealTrenpm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.

added 1287 packages, and audited 1288 packages in 19s

279 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
💿 That's it! `cd` into "/private/tmp/my-remix-app" and check the README for development and deploy instructions!
$ cd my-remix-app/
$ npm run typecheck

> typecheck
> tsc

node_modules/@remix-run/architect/dist/server.d.ts:3:135 - error TS2307: Cannot find module 'aws-lambda' or its corresponding type declarations.

3 import type { APIGatewayProxyEventHeaders, APIGatewayProxyEventV2, APIGatewayProxyHandlerV2, APIGatewayProxyStructuredResultV2 } from "aws-lambda";
                                                                                                                                        ~~~~~~~~~~~~

node_modules/@remix-run/node/dist/upload/fileUploadHandler.d.ts:47:22 - error TS2420: Class 'NodeOnDiskFile' incorrectly implements interface 'File'.
  Property 'prototype' is missing in type 'NodeOnDiskFile' but required in type 'File'.

47 export declare class NodeOnDiskFile implements File {
                        ~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2591:5
    2591     prototype: Blob;
             ~~~~~~~~~
    'prototype' is declared here.

node_modules/@remix-run/server-runtime/dist/cookies.d.ts:1:65 - error TS7016: Could not find a declaration file for module 'cookie'. '/private/tmp/my-remix-app/node_modules/@remix-run/server-runtime/node_modules/cookie/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/cookie` if it exists or add a new declaration (.d.ts) file containing `declare module 'cookie';`

1 import type { CookieParseOptions, CookieSerializeOptions } from "cookie";
                                                                  ~~~~~~~~

node_modules/@remix-run/server-runtime/dist/sessions.d.ts:1:65 - error TS7016: Could not find a declaration file for module 'cookie'. '/private/tmp/my-remix-app/node_modules/@remix-run/server-runtime/node_modules/cookie/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/cookie` if it exists or add a new declaration (.d.ts) file containing `declare module 'cookie';`

1 import type { CookieParseOptions, CookieSerializeOptions } from "cookie";
                                                                  ~~~~~~~~

node_modules/@types/glob/index.d.ts:29:42 - error TS2694: Namespace '"/private/tmp/my-remix-app/node_modules/minimatch/dist/cjs/index"' has no exported member 'IOptions'.

29     interface IOptions extends minimatch.IOptions {
                                            ~~~~~~~~

node_modules/@types/glob/index.d.ts:74:30 - error TS2724: '"/private/tmp/my-remix-app/node_modules/minimatch/dist/cjs/index"' has no exported member named 'IMinimatch'. Did you mean 'Minimatch'?

74         minimatch: minimatch.IMinimatch;
                                ~~~~~~~~~~


Found 6 errors in 5 files.

Errors  Files
     1  node_modules/@remix-run/architect/dist/server.d.ts:3
     1  node_modules/@remix-run/node/dist/upload/fileUploadHandler.d.ts:47
     1  node_modules/@remix-run/server-runtime/dist/cookies.d.ts:1
     1  node_modules/@remix-run/server-runtime/dist/sessions.d.ts:1
     2  node_modules/@types/glob/index.d.ts:29
$ 
```

**Comment by lpsinger** [comment]
#6275 is related, but there are a number of other problems evident here too --- several of them traceable to #6051.

**Comment by dani-mp** [comment]
Is there a workaround for this? I'm having the same problem in Vercel when I try to upgrade my Remix app to 1.16.X.

**Comment by machour** [comment]
Related (duplicate?) of #4371 but for yet another template

**Comment by MichaelDeBoey** [comment]
Also related to #6244

**Comment by quangv** [comment]
Believe it's also related to #6403

**Comment by MichaelDeBoey** [comment]
This is now fixed by #6108 and will be released in v2

**Comment by tommyasai** [comment]
V2 release has fixed some errors but didn't fix the following two errors. Does anyone see the same issue?

There's a [StackOverflow discussion](https://stackoverflow.com/questions/75890950/node-modules-minimatch-dist-cjs-index-has-no-exported-member-ioptions) that addresses this issue, but I am not sure how to resolve it in my project.

```
node_modules/@types/glob/index.d.ts:29:42 - error TS2694: Namespace '"/private/tmp/my-remix-app/node_modules/minimatch/dist/cjs/index"' has no exported member 'IOptions'.

29     interface IOptions extends minimatch.IOptions {
                                            ~~~~~~~~

node_modules/@types/glob/index.d.ts:74:30 - error TS2724: '"/private/tmp/my-remix-app/node_modules/minimatch/dist/cjs/index"' has no exported member named 'IMinimatch'. Did you mean 'Minimatch'?

74         minimatch: minimatch.IMinimatch;
```

**Comment by MichaelDeBoey** [comment]
@tommyasai That error is fixed by #7317

**Comment by tommyasai** [comment]
@MichaelDeBoey Confirmed it works. I was using 2.0.0-pre.3. Thank you.
